### PR TITLE
feat: Add error handling/form validation to custom version

### LINKED ISSUE #226: Handle CORS errors nicely when loading a custom version of EDAM
Current behaviour
==============

Whenever a user loads a custom version of EDAM, if there is an error while loading the URL the UI will give no error message and no feedback:
image

Suggested improvements
===================

* Display an error message when an error occurs while loading the file, then go back to the previous version selected.
* If possible, the error message should be specific and readable, e.g. explain if this is a CORS error, or if the file doesn't seem to be in the right format.

### PR DESCRIPTION
**Pull request by HagerDakroury**
### Checklist

- [x] I indicated which issue (if any) is closed with this PR using [closing keywords](https://help.github.com/articles/closing-issues-using-keywords)
- [x] I only changed lines related to my PR (no bulk reformatting)
- [x] I indicated the source and check the license if I re-use code, or I did not re-use code
- [x] I made my best to solve only one issue in this PR, or explain why multi had to be solved at once.

### Issue

related-to #226 

### Details

1. Cleaned and commented loadTree code and removed redundant code
2. Added error handling when loading a custom version
corslow

3. Added form validation (to detect wrong format)
validaitonlow

### CODE REVIEW

File: js/tree-edam-stand-alone.js [hunk 1]
```diff
@@ -286,13 +274,21 @@ function interactive_edam_browser(){
             .toLowerCase()
             .replace(/[-]+/g,'-');
             let params = ""
+            //update url only if the tree is loaded successfully
+            if(getCookie("isLoaded",true)=="true") {
```
Comment (bryan-brancotte) [inline js/tree-edam-stand-alone.js:278]: good idea ! I was really annoying to get stuck in an invalide version

**Review by hmenager — APPROVED** [review]
looks good to me!

**Review by bryan-brancotte — COMMENTED** [review]
There is not "Request help" option for the PR ;) I pulled the cors branch, but when loading `https://github.com/edamontology/edamontology/releases/download/1.25/EDAM_1.25.owl` I don't get any error message. I both Chrome and Firefox I get this error in the console. Am I missing something ?
Capture d’écran de 2021-08-05 10-25-21

**Comment by HagerDakroury** [comment]
> There is not "Request help" option for the PR ;) I pulled the cors branch, but when loading `https://github.com/edamontology/edamontology/releases/download/1.25/EDAM_1.25.owl` I don't get any error message. I both Chrome and Firefox I get this error in the console. Am I missing something ?
> Capture d’écran de 2021-08-05 10-25-21

maybe you need to do an `npm install` first? (since I updated edam2json-js version in the process)

**Comment by hmenager** [comment]
> > There is not "Request help" option for the PR ;) I pulled the cors branch, but when loading `https://github.com/edamontology/edamontology/releases/download/1.25/EDAM_1.25.owl` I don't get any error message. I both Chrome and Firefox I get this error in the console. Am I missing something ?
> > Capture d’écran de 2021-08-05 10-25-21
> 
> maybe you need to do an `npm install` first? (since I updated edam2json-js version in the process)

It worked for me when testing, but I indeed first ran `npm install`.

**Review by bryan-brancotte — APPROVED** [review]
Ok, it was simply the npm install missing... works fine for me now !

**Pull request merged**